sygnal/gcmpushkin.py

```diff
@@ -347,16 +347,24 @@ async def _dispatch_notification_unlimited(
         with self.sygnal.tracer.start_span(
             "gcm_dispatch", tags=span_tags, child_of=context.opentracing_span
         ) as span_parent:
+            # TODO: Implement collapse_key to queue only one message per room.
+            failed: List[str] = []
+
             data = GcmPushkin._build_data(n, device)
+
+            # Reject pushkey if default_payload is misconfigured
+            if not data:
+                log.error(
+                    "Default_payload was misconfigured, this value must be a dict."
+                )
```

Comment:
I'd be tempted to log in `_build_data` if possible, so that the log message is next to the logic that detects the error condition.